src/circleci/test/retest.clj

```diff
@@ -0,0 +1,23 @@
+(ns circleci.test.retest
+  (:require [clojure.data.xml :as xml]
+            [clojure.java.io :as io]
+            [circleci.test :as test]))
+
+(defn- failed-tests-from [root]
+  (let [suite (:name (:attrs root))]
+    (for [test-case (:content root)
+          :when (some #(= :failure (:tag %)) (:content test-case))]
+      (symbol suite (:name (:attrs test-case))))))
+
+(defn- failed-tests [report-dir]
+  (when-not (.exists (io/file report-dir))
```

Comment:
It occurs to me that we can assume if `-main` is called it won't be interactively, but I should add a public non-main function for repl use.